### PR TITLE
Fix dev

### PR DESCRIPTION
**Pull request by OverLordGoldDragon**
This fixes dev and passes all 1D tests. But it goes further:

 1. Removed `Modulus` implementations. The cited purpose is "stability" and "not returning NaNs for gradients", but `torch.abs(x)` accomplishes the same AFAIK, and `test_modulus()` passes.
 2. Dropped trailing dim everywhere (i.e. `view_as_real`) except `subsample_fourier` since complex `mean()` isn't implemented in 1.7. I'm not aware of its need beyond modulus; it only complicates code.

### Todo

 - [ ] Docs (will change after review)
 - [ ] 2D & 3D (I leave this to devs)

### CODE REVIEW

**Comment by lostanlen** [comment]
many thanks @OverLordGoldDragon 
if i understand correctly this is the first PR scheduled to be merged to dev?

File: kymatio/backend/torch_backend.py [hunk 1]
```diff
@@ -128,21 +33,20 @@ def contiguous_check(x):
...
-        cls.complex_contiguous_check(x)
-        norm = ModulusStable.apply(x)[..., None]
-        return norm
+        cls.complex_check(x)
+        return torch.abs(x)
 
     @staticmethod
-    def concatenate(arrays, dim=2):
-        return torch.stack(arrays, dim=dim)
+    def concatenate(arrays, axis=-2):
```
Comment (MuawizChaudhary) [inline kymatio/backend/torch_backend.py:48]: why the arbitary change in name?

File: kymatio/backend/torch_backend.py [hunk 2]
```diff
@@ -185,8 +89,10 @@ def cdgmm(cls, A, B):
 
         cls.complex_contiguous_check(A)
 
-        if A.shape[-len(B.shape):-1] != B.shape[:-1]:
-            raise RuntimeError('The filters are not compatible for multiplication.')
+        if A.shape[-B.ndim:] != B.shape:
```
Comment (MuawizChaudhary) [inline kymatio/backend/torch_backend.py:92]: Why?

File: kymatio/scattering1d/backend/torch_backend.py [hunk 3]
```diff
@@ -1,18 +1,8 @@
...
-
-if version.parse(torch.__version__) >= version.parse('1.8'):
-    _fft = lambda x: torch.view_as_real(torch.fft.fft(torch.view_as_complex(x)))
-    _ifft = lambda x: torch.view_as_real(torch.fft.ifft(torch.view_as_complex(x)))
-    _irfft = lambda x: torch.fft.ifft(torch.view_as_complex(x)).real[..., None]
-else:
-    _fft = lambda x: torch.fft(x, 1, normalized=False)
-    _ifft = lambda x: torch.ifft(x, 1, normalized=False)
-    _irfft = lambda x: torch.irfft(x, 1, normalized=False, onesided=False)[..., None]
-
```
[the remark this answers is not in the document]
Comment (MuawizChaudhary) [reply kymatio/scattering1d/backend/torch_backend.py:15]: @janden

File: tests/scattering1d/test_torch_backend_1d.py [hunk 4]
```diff
@@ -93,7 +93,7 @@ def test_modulus(device, backend, random_state=42):
     """
     Tests the stability and differentiability of modulus
     """
-    if backend.name.endswith('_skcuda') and device == "cpu":
+    if backend.name == "torch_skcuda" and device == "cpu":
```
Comment (MuawizChaudhary) [inline tests/scattering1d/test_torch_backend_1d.py:96]: Why?

**Review by MuawizChaudhary — COMMENTED** [review]
If we're removing sk_cuda and the associated modulus stable class, we should extend this to the other dimentions of scattering in this pull request.

**Comment by MuawizChaudhary** [comment]
Breaks backwards compatibility
